sdk/core/core/inc/az_span.h

```diff
@@ -63,34 +71,62 @@ az_span_sub(az_span const span, size_t const begin, size_t const end) {
  * Returns `true` if a content of the @a span is equal to a content of the @b
  * span.
  */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_content(az_span const a, az_span const b) {
+  return a._internal.length == b._internal.length
+      && memcmp(a._internal.begin, b._internal.begin, a._internal.length) == 0;
+}
+
+/**
+ * Returns `true` if a capacity of the @a span is equal to a capacity of the @b
+ * span.
+ */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_capacity(az_span const a, az_span const b) {
+  return a._internal.capacity == b._internal.capacity;
+}
+
+/**
+ * Returns `true` if span @b a is equal span @b b on content and capacity
+ */
 AZ_NODISCARD AZ_INLINE bool az_span_is_equal(az_span const a, az_span const b) {
```

Comment:
Do we need this?

Reply:
it is used from many places currently. I will move it to private for now